vendor/github.com/docker/go-connections/tlsconfig/config.go

```diff
@@ -83,6 +88,65 @@ func certPool(caFile string) (*x509.CertPool, error) {
 	return certPool, nil
 }
 
+func IsErrEncryptedKey(err error) bool {
+	return errors.Cause(err) == errCantDecrypt
+}
+
+// getPrivateKey returns the private key in 'keyBytes', in PEM-encoded format.
+// If the private key is encrypted, 'passphrase' is used to decrypted the
+// private key.
+func getPrivateKey(keyBytes []byte, passphrase string) ([]byte, error) {
+	// this section makes some small changes to code from notary/tuf/utils/x509.go
+	pemBlock, _ := pem.Decode(keyBytes)
+	if pemBlock == nil {
+		return nil, fmt.Errorf("no valid private key found")
+	}
+
+	var err error
+	if x509.IsEncryptedPEMBlock(pemBlock) {
+		keyBytes, err = x509.DecryptPEMBlock(pemBlock, []byte(passphrase))
+		if err != nil {
+			return nil, errCantDecrypt
```

Comment:
That's true - we can probably just pass the error returned by `DecryptPEMBlock`, maybe wrapped like `errors.Wrap(err, "private key is encrypted")`.  The docs from `DecryptPEMBlock` say:

> DecryptPEMBlock takes a password encrypted PEM block and the password used to encrypt it and returns a slice of decrypted DER encoded bytes. It inspects the DEK-Info header to determine the algorithm used for decryption. If no DEK-Info header is present, an error is returned. If an incorrect password is detected an IncorrectPasswordError is returned. Because of deficiencies in the encrypted-PEM format, it's not always possible to detect an incorrect password. In these cases no error will be returned but the decrypted DER bytes will be random noise.

So in `IsErrEncryptedKey` maybe we can just check `errors.Cause(err) == x509.IncorrectPasswordError`?